src/Moq/ExpressionExtensions.cs

```diff
@@ -153,9 +157,25 @@ void Split(Expression e, out Expression r /* remainder */, out LambdaExpressionP
 							arguments[ai] = lhs.Arguments[ai];
 						}
 						arguments[arguments.Length - 1] = assignmentExpression.Right;
+
+						Func<Expression, Expression, BinaryExpression> assign;
+						switch (e.NodeType)
```

Comment:
This isn't particularly nice. Instead of `Expression.Assign`, use `Expression.MakeBinary` and specify `e.NodeType`.